language/diem-framework/modules/PaymentScripts.move

```diff
@@ -71,10 +74,91 @@ module PaymentScripts {
+    /// | ----------------           | --------------                                   | -------------                                                                                                                       |
+    /// | `Errors::NOT_PUBLISHED`    | `DiemAccount::EPAYER_DOESNT_HOLD_CURRENCY`       | `payer` doesn't hold a balance in `Currency`.                                                                                       |
+    /// | `Errors::LIMIT_EXCEEDED`   | `DiemAccount::EINSUFFICIENT_BALANCE`             | `amount` is greater than `payer`'s balance in `Currency`.                                                                           |
+    /// | `Errors::INVALID_ARGUMENT` | `DiemAccount::ECOIN_DEPOSIT_IS_ZERO`             | `amount` is zero.                                                                                                                   |
+    /// | `Errors::NOT_PUBLISHED`    | `DiemAccount::EPAYEE_DOES_NOT_EXIST`             | No account exists at the `payee` address.                                                                                           |
+    /// | `Errors::INVALID_ARGUMENT` | `DiemAccount::EPAYEE_CANT_ACCEPT_CURRENCY_TYPE`  | An account exists at `payee`, but it does not accept payments in `Currency`.                                                        |
+    /// | `Errors::INVALID_STATE`    | `AccountFreezing::EACCOUNT_FROZEN`               | The `payee` account is frozen.                                                                                                      |
+    /// | `Errors::LIMIT_EXCEEDED`   | `DiemAccount::EWITHDRAWAL_EXCEEDS_LIMITS`        | `payer` has exceeded its daily withdrawal limits for the backing coins of XDX.                                                      |
+    /// | `Errors::LIMIT_EXCEEDED`   | `DiemAccount::EDEPOSIT_EXCEEDS_LIMITS`           | `payee` has exceeded its daily deposit limits for XDX.                                                                              |
+    ///
+    /// # Related Scripts
+    /// * `AccountCreationScripts::create_child_vasp_account`
+    /// * `AccountCreationScripts::create_parent_vasp_account`
+    /// * `AccountAdministrationScripts::add_currency_to_account`
+    /// * `PaymentScripts::peer_to_peer_with_metadata`
+    public(script) fun peer_to_peer_by_signers<Currency: store>(
+        payer: signer,
+        payee: signer,
+        amount: u64,
+        metadata: vector<u8>,
+    ) {
+        let payer_withdrawal_cap = DiemAccount::extract_withdraw_capability(&payer);
+        DiemAccount::pay_by_signers<Currency>(
+            &payer_withdrawal_cap, &payee, amount, metadata
+        );
+        DiemAccount::restore_withdraw_capability(payer_withdrawal_cap);
+    }
+
     spec peer_to_peer_with_metadata {
-        use 0x1::Signer;
+        /// TODO(timeout): this currently times out
```

Comment:
This timeout is interesting because we haven't seen prover timeouts in a while since monomorphization was added to prover. I will share this code with the prover team.

Reply:
I didn't add this todo comment. It's there in current payment script